target/linux/ath79/files/arch/mips/include/asm/mach-ath79/rb4xx_cpld.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPI driver definitions for the CPLD chip on the Mikrotik RB4xx boards
+ *
+ * Copyright (C) 2010 Gabor Juhos <juhosg@openwrt.org>
+ *
+ * This file was based on the patches for Linux 2.6.27.39 published by
+ * MikroTik for their RouterBoard 4xx series devices.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation.
+ */
+
+#define CPLD_GPIO_LED1		0
+#define CPLD_GPIO_LED2		1
+#define CPLD_GPIO_LED3		2
+#define CPLD_GPIO_LED4		3
+#define CPLD_GPIO_FAN		4
+#define CPLD_GPIO_ALE		5
+#define CPLD_GPIO_CLE		6
+#define CPLD_GPIO_nCE		7
+#define CPLD_GPIO_LED5		8
+
+#define CPLD_NUM_GPIOS		9
```

Comment:
Just a very quick comment before I take a deeper look (later): I feel these are really describing the hardware and as such I wonder if they shouldn't belong to the DTS rather than a header. This would possibly make the driver more generic and adaptable to other kinds of hardware.
Otherwise this looks like it's going in the right direction, thanks for moving forward!

Reply:
Actually, the only usage of these GPIO constants in the code is in the CPLD driver (uses CPLD_CFG_LED5, CPLD_CMD_LED5_ON/OFF), so it's true most of them can be purged if that's better.

N.B. I believe the pins are actually hard coded in the CPLD firmware, so you can't change them.

Reply:
My point isn't so much about the ability to change the pins as the ability to adapt the driver to other hardware implementations. I was under the impression that anything that "describes" the hardware should go in DTS. But I'm no expert on this matter :)

Reply:
I have moved all these defines into the .c file, so welcome further comment on if these should simply be deleted - or left in the source to document what the pins actually do